ld/src/ld/parsers/archive_file.cpp

```diff
@@ -536,6 +537,17 @@ bool File<A>::forEachAtom(ld::File::AtomHandler& handler) const
 	return didSome;
 }
 
+template <typename A>
+bool File<A>::hasSwift() const {
+	for (const auto& entry : _hashTable) {
```

Comment:
what perf cost do you see for this?

Reply:
e.g., if you add a CACurrentMediaTime at the start and end of this function, and then sum up all the times, what % of total time is it?

Reply:
The cost is within measurement noise for our codebase. I profiled linking our largest app before and after applying this PR, I got average link times of 38.37s vs 39.24s with a variance of > 1s for each run.

Reply:
fair enough